services/src/main/scala/cromwell/services/metadata/impl/WriteMetadataActor.scala

```diff
@@ -34,6 +39,8 @@ class WriteMetadataActor(override val batchSize: Int,
     val allPutEvents: Iterable[MetadataEvent] = putWithoutResponse ++ putWithResponse.flatMap(_._1)
     val dbAction = addMetadataEvents(allPutEvents)
 
+    statsRecorder.processEvents(allPutEvents) foreach(a => log.warning(s"${a.workflowId} has logged a heavy amount of metadata (${a.count} rows)"))
```

Comment:
I don't know off the top of my head what our `batchSize` is (line 20) but it seems like this could be a lot of logging if we do it every batch.

Reply:
Maybe renaming the method to `processEventsAndEmitWarnings` would make it more clear what's going on?